zeropb.go

```diff
@@ -58,6 +215,95 @@ func SetUint64(buf *[]byte, offsets *FastIntMap, fieldID int, x uint64) {
+func SetZigZagInt64(buf *[]byte, offsets *FastIntMap, fieldID int, x int64) {
+	var scratch [binary.MaxVarintLen64]byte
+	n := binary.PutVarint(scratch[:], x)
+	var valLen, val []byte = nil, scratch[:n]
+	setSingle(buf, offsets, fieldID, proto.WireVarint, valLen, val)
+}
+
+// SetFixedUint32 set a little-endian encoded uint32 field with the given field
+// id, overwriting an old value if present.
+func SetFixedUint32(buf *[]byte, offsets *FastIntMap, fieldID int, x uint32) {
+	var scratch [4]byte
+	binary.LittleEndian.PutUint32(scratch[:], x)
+	var valLen, val []byte = nil, scratch[:]
+	setSingle(buf, offsets, fieldID, proto.WireFixed32, valLen, val)
+}
+
+// SetFixedUint64 set a little-endian encoded uint64 field with the given field
+// id, overwriting an old value if present.
+func SetFixedUint64(buf *[]byte, offsets *FastIntMap, fieldID int, x uint64) {
+	var scratch [8]byte
+	binary.LittleEndian.PutUint64(scratch[:], x)
+	var valLen, val []byte = nil, scratch[:]
+	setSingle(buf, offsets, fieldID, proto.WireFixed64, valLen, val)
+}
+
+// SetFixedInt32 set a little-endian encoded int32 field with the given field
+// id, overwriting an old value if present.
+func SetFixedInt32(buf *[]byte, offsets *FastIntMap, fieldID int, x int32) {
+	var scratch [4]byte
+	binary.LittleEndian.PutUint32(scratch[:], uint32(x))
+	var valLen, val []byte = nil, scratch[:]
+	setSingle(buf, offsets, fieldID, proto.WireFixed32, valLen, val)
+}
+
+// SetFixedInt64 set a little-endian encoded int64 field with the given field
+// id, overwriting an old value if present.
+func SetFixedInt64(buf *[]byte, offsets *FastIntMap, fieldID int, x int64) {
+	var scratch [8]byte
+	binary.LittleEndian.PutUint64(scratch[:], uint64(x))
+	var valLen, val []byte = nil, scratch[:]
+	setSingle(buf, offsets, fieldID, proto.WireFixed64, valLen, val)
+}
+
+// SetFloat32 set a little-endian, IEEE 754 encoded float32 field with the given
+// field id, overwriting an old value if present.
+func SetFloat32(buf *[]byte, offsets *FastIntMap, fieldID int, x float32) {
+	var scratch [4]byte
+	bits := math.Float32bits(x)
+	binary.LittleEndian.PutUint32(scratch[:], bits)
+	var valLen, val []byte = nil, scratch[:]
+	setSingle(buf, offsets, fieldID, proto.WireFixed32, valLen, val)
+}
+
+// SetFloat64 set a little-endian, IEEE 754 encoded float64 field with the given
+// field id, overwriting an old value if present.
+func SetFloat64(buf *[]byte, offsets *FastIntMap, fieldID int, x float64) {
+	var scratch [8]byte
+	bits := math.Float64bits(x)
+	binary.LittleEndian.PutUint64(scratch[:], bits)
+	var valLen, val []byte = nil, scratch[:]
+	setSingle(buf, offsets, fieldID, proto.WireFixed64, valLen, val)
+}
+
+// SetString set an encoded string field with the given field id, overwriting an
+// old value if present.
+func SetString(buf *[]byte, offsets *FastIntMap, fieldID int, x string) {
+	var scratch [binary.MaxVarintLen64]byte
+	hdr := *(*reflect.StringHeader)(unsafe.Pointer(&x))
```

Comment:
@nvanbenschoten Is there some better way to do this? I have the usual `[]byte` -> `string` trick above in `GetString` but I can't recall what it is for the other way: `string` -> `[]byte`